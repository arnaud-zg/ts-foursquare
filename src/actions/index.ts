import { TLifeAction } from './life'
import { TVenuesAction } from './venues'

export type TRootActionType = TVenuesAction | TLifeAction

export type TRootAction = TVenuesAction | TLifeAction

export * from './life'
export * from './venues'
