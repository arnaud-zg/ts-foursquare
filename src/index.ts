require('isomorphic-fetch')

export * from './standalone'
export * from './utils'

export * from './DEPRECATED_actions'
export * from './DEPRECATED_reducers'
export * from './DEPRECATED_selectors'
