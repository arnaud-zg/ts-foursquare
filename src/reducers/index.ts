import { NVenues } from '../../types/venues'

export type TRootState = {
  venues: NVenues.IVenuesState
}
