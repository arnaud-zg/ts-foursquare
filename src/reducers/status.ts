import { createReducer, getType } from 'typesafe-actions'
import { NStatus } from '../../types'
import {
  getListsAsync,
  getPhotosDetailsAsync,
  getVenuesCategoriesAsync,
  getVenuesExploreAsync,
  getVenuesLikesAsync,
  getVenuesListedAsync,
  getVenuesNextVenuesAsync,
  getVenuesSearchAsync,
  getVenuesSimilarAsync,
  getVenuesSuggestCompletionAsync,
  getVenuesTrendingAsync,
  TRootAction,
} from '../actions'
import { ASYNC_ACTION_NAME_MAPPING } from '../constants/asyncAction'

export const initialState: NStatus.IState = {}

export const statusReducer = createReducer<NStatus.IState, TRootAction>(
  initialState
)
  .handleAction(
    [
      getType(getListsAsync.request),
      getType(getPhotosDetailsAsync.request),
      getType(getVenuesCategoriesAsync.request),
      getType(getVenuesExploreAsync.request),
      getType(getVenuesLikesAsync.request),
      getType(getVenuesListedAsync.request),
      getType(getVenuesNextVenuesAsync.request),
      getType(getVenuesSearchAsync.request),
      getType(getVenuesSimilarAsync.request),
      getType(getVenuesSuggestCompletionAsync.request),
      getType(getVenuesTrendingAsync.request),
    ],
    (state, action: TRootAction): NStatus.IState => ({
      ...state,
      [ASYNC_ACTION_NAME_MAPPING[action.type]]: {
        ...state[action.type],
        hasError: false,
        isCancelled: false,
        isLoading: true,
      },
    })
  )
  .handleAction(
    [
      getType(getListsAsync.failure),
      getType(getPhotosDetailsAsync.failure),
      getType(getVenuesCategoriesAsync.failure),
      getType(getVenuesExploreAsync.failure),
      getType(getVenuesLikesAsync.failure),
      getType(getVenuesListedAsync.failure),
      getType(getVenuesNextVenuesAsync.failure),
      getType(getVenuesSearchAsync.failure),
      getType(getVenuesSimilarAsync.failure),
      getType(getVenuesSuggestCompletionAsync.failure),
      getType(getVenuesTrendingAsync.failure),
    ],
    (state, action: TRootAction): NStatus.IState => ({
      ...state,
      [ASYNC_ACTION_NAME_MAPPING[action.type]]: {
        ...state[action.type],
        hasError: true,
        isCancelled: false,
        isLoading: false,
      },
    })
  )
  .handleAction(
    [
      getType(getListsAsync.cancel),
      getType(getPhotosDetailsAsync.cancel),
      getType(getVenuesCategoriesAsync.cancel),
      getType(getVenuesExploreAsync.cancel),
      getType(getVenuesLikesAsync.cancel),
      getType(getVenuesListedAsync.cancel),
      getType(getVenuesNextVenuesAsync.cancel),
      getType(getVenuesSearchAsync.cancel),
      getType(getVenuesSimilarAsync.cancel),
      getType(getVenuesSuggestCompletionAsync.cancel),
      getType(getVenuesTrendingAsync.cancel),
    ],
    (state, action: TRootAction): NStatus.IState => ({
      ...state,
      [ASYNC_ACTION_NAME_MAPPING[action.type]]: {
        ...state[action.type],
        hasError: false,
        isCancelled: true,
        isLoading: false,
      },
    })
  )
  .handleAction(
    [
      getType(getListsAsync.success),
      getType(getPhotosDetailsAsync.success),
      getType(getVenuesCategoriesAsync.success),
      getType(getVenuesExploreAsync.success),
      getType(getVenuesLikesAsync.success),
      getType(getVenuesListedAsync.success),
      getType(getVenuesNextVenuesAsync.success),
      getType(getVenuesSearchAsync.success),
      getType(getVenuesSimilarAsync.success),
      getType(getVenuesSuggestCompletionAsync.success),
      getType(getVenuesTrendingAsync.success),
    ],
    (state, action: TRootAction): NStatus.IState => ({
      ...state,
      [ASYNC_ACTION_NAME_MAPPING[action.type]]: {
        ...state[action.type],
        hasError: false,
        isCancelled: false,
        isLoading: false,
      },
    })
  )
