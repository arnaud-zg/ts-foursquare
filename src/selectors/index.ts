export * from './life'
export * from './lists'
export * from './photos'
export * from './request'
export * from './status'
export * from './venues'
