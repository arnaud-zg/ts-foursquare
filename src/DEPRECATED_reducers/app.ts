import { combineReducers } from 'redux'
import { initialState as initialStateLife, lifeReducer } from './life'
import { initialState as initialStateLists, listsReducer } from './lists'
import { initialState as initialStatePhotos, photosReducer } from './photos'
import { initialState as initialStateRequest, requestReducer } from './request'
import { initialState as initialStateStatus, statusReducer } from './status'
import { initialState as initialStateVenues, venuesReducer } from './venues'

export const initialState = {
  life: initialStateLife,
  lists: initialStateLists,
  photos: initialStatePhotos,
  request: initialStateRequest,
  status: initialStateStatus,
  venues: initialStateVenues,
}

export type TState = typeof initialState

export const appReducer = combineReducers({
  life: lifeReducer,
  lists: listsReducer,
  photos: photosReducer,
  request: requestReducer,
  status: statusReducer,
  venues: venuesReducer,
})
