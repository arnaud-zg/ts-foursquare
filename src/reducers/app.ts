import { combineReducers } from 'redux'
import { initialState as initialStateLife } from '../../src/reducers/life'
import { initialState as initialStateStatus } from '../../src/reducers/status'
import { initialState as initialStateVenues } from '../../src/reducers/venues'
import { lifeReducer } from './life'
import { statusReducer } from './status'
import { venuesReducer } from './venues'

export const initialState = {
  life: initialStateLife,
  status: initialStateStatus,
  venues: initialStateVenues,
}

export type TState = typeof initialState

export const appReducer = combineReducers({
  life: lifeReducer,
  status: statusReducer,
  venues: venuesReducer,
})
