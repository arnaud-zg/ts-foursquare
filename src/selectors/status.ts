import { NRoot } from '../../types/root.d'
import { NStatus } from '../../types/status'
import { TRootActionType } from '../actions'

export const statusSelector = (state: NRoot.IState) => state.status

export const moduleStateSelector = (
  state: NRoot.IState,
  moduleKey: TRootActionType
): NStatus.IModuleState =>
  state.status[moduleKey] || {
    hasError: false,
    isLoading: false,
  }
