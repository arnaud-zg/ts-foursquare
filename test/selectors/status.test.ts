import { EVenuesAction } from '../../src/actions'
import { initialState } from '../../src/reducers/app'
import { moduleStateSelector, statusSelector } from '../../src/selectors/status'
import { NStore } from '../../types/store'

const state: NStore.IState = {
  ...initialState,
  status: {
    [EVenuesAction.GET_VENUES_SEARCH_REQUEST]: {
      hasError: false,
      isCancelled: false,
      isLoading: false,
    },
  },
}

describe('selectors/status', () => {
  it('should get status', () => {
    expect(statusSelector(state)).toMatchSnapshot()
  })

  it('should get module state', () => {
    expect(
      moduleStateSelector(initialState, EVenuesAction.GET_VENUES_SEARCH_REQUEST)
    ).toMatchSnapshot()

    expect(
      moduleStateSelector(state, EVenuesAction.GET_VENUES_SEARCH_REQUEST)
    ).toMatchSnapshot()
  })
})
