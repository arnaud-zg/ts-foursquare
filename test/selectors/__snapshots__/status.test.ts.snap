// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`selectors/status should get module state 1`] = `
Object {
  "hasError": false,
  "isLoading": false,
}
`;

exports[`selectors/status should get module state 2`] = `
Object {
  "hasError": false,
  "isLoading": false,
}
`;

exports[`selectors/status should get status 1`] = `
Object {
  "GET_VENUES_SEARCH_REQUEST": Object {
    "hasError": false,
    "isLoading": false,
  },
}
`;
